src/app/home-page/components/slideshow/slideshow.component.spec.ts

```diff
@@ -24,62 +30,54 @@ describe('SlideshowComponent', () => {
+    component.intervalStart = 74;
+    component.runAutoplay();
+    expect(component.intervalStart).toBe(76);
   });
 
-  it('should return undefined if we call stop', () => {
-    fixture = TestBed.createComponent(SlideshowComponent);
-    component = fixture.componentInstance;
-    functionsReturn = component.continue();
-    expect(functionsReturn).toBeUndefined();
-  });
+  it('should check equality if moveSlide method is called', () => {
+    component.selectedSlideIndex = 1;
+    component.currentSlideIndex = 1;
+    component.moveSlide(SLIDE_DIRECTION.RIGHT);
+    expect(component.selectedSlideIndex).toBe(component.currentSlideIndex);
 
-  it('govno', () => {
-    component.isStoped = true;
-    component.continue();
-    expect(component.isHovered).toBe(false);
+    component.selectedSlideIndex = 1;
+    component.currentSlideIndex = 1;
+    component.moveSlide(SLIDE_DIRECTION.LEFT);
+    expect(component.selectedSlideIndex).toBe(component.currentSlideIndex);
   });
 
-  it('govno2', () => {
-    component.isStoped = true;
-    component.pause();
-    expect(component.isHovered).toBe(false);
+  it('should check currentSlideIndex ', () => {
+    component.currentSlideIndex = 0;
+    component.slidesLength = 1;
+    component.moveSlide(SLIDE_DIRECTION.RIGHT);
+    expect(component.currentSlideIndex).toBe(0);
   });
 
-  it('govno3', () => {
-    component.moveSlide(1);
-    expect(component.selectedSlideIndex).toBe(1);
+  it('should set isHovered to true when pause method is called', () => {
+    component.isStoped = false;
+    component.pause();
+    expect(component.isHovered).toBe(true);
   });
 
-  it('govno4', () => {
-    component.moveSlide(0);
-    const slidesLength = component.responseImgs.length;
-    expect(component.selectedSlideIndex).toBe(slidesLength - 1);
+  it('should return undefined when pause method is called and isStopped === true', () => {
+    component.isStoped = true;
+    component.pause();
+    expect(component.pause()).toBeUndefined();
   });
 
-  it('should return undefined if we call stop', () => {
-    fixture = TestBed.createComponent(SlideshowComponent);
-    component = fixture.componentInstance;
-    functionsReturn = component.pause();
-    expect(functionsReturn).toBeUndefined();
+  it('should call runAutoplay method when continue method is called', () => {
+    component.isStoped = false;
+    const spy = spyOn(component, 'runAutoplay');
+    component.continue();
+    expect(spy).toHaveBeenCalled();
   });
 
-  it('should return undefined if we call stop', () => {
-    fixture = TestBed.createComponent(SlideshowComponent);
-    component = fixture.componentInstance;
-    functionsReturn = component.moveSlide(1);
-    expect(functionsReturn).toBeUndefined();
+  it('should return undefined when continue method is called and isStopped === true', () => {
+    component.isStoped = true;
+    component.continue();
+    expect(component.continue()).toBeUndefined();
```

Comment:
all these methods (pause/continue) return undefined
check the properties they change, refactor the rest one with toBeUndefined checks

see example:
```
component.isStoped = true;
component.pause();
expect(this.isHovered).toBe(false) 
```